pkg/healthcheck/healthcheck_output.go

```diff
@@ -160,49 +279,62 @@ func RunExtensionsChecks(wout io.Writer, werr io.Writer, extensions []string, fl
 		}
 
 		if err == nil {
-			if isatty.IsTerminal(os.Stdout.Fd()) {
-				spin.Suffix = fmt.Sprintf(" Running %s extension check", extension)
-				spin.Color("bold") // this calls spin.Restart()
-			}
-			// Path is constructed from the switch statements above and will
-			// be a valid Linkerd subcommand.
-			//nolint:gosec
-			plugin := exec.Command(path, args...)
-			var stdout, stderr bytes.Buffer
-			plugin.Stdout = &stdout
-			plugin.Stderr = &stderr
-			plugin.Run()
-			extensionResults, err := parseJSONCheckOutput(stdout.Bytes())
-			spin.Stop()
-			if err != nil {
-				command := fmt.Sprintf("%s %s", path, strings.Join(args, " "))
-				if len(stderr.String()) > 0 {
-					err = errors.New(stderr.String())
-				} else {
-					err = fmt.Errorf("invalid extension check output from \"%s\" (JSON object expected):\n%s\n[%w]", command, stdout.String(), err)
-				}
-				results.Results = append(results.Results, CheckResult{
-					Category:    CategoryID(extensionCmd),
-					Description: fmt.Sprintf("Running: %s", command),
-					Err:         err,
-					HintURL:     HintBaseURL(version.Version) + "extensions",
-				})
+			checkResults, checkSuccess := runCheck(path, args, extension, CategoryID(extensionCmd), spin)
+			if !checkSuccess {
 				success = false
-			} else {
-				results.Results = append(results.Results, extensionResults.Results...)
 			}
+			results.Results = append(results.Results, checkResults...)
 		}
 
-		var extensionSuccess bool
-		extensionSuccess, warning = RunChecks(wout, werr, results, output)
+		extensionSuccess, extensionWarning := RunChecks(wout, werr, results, output)
 		if !extensionSuccess {
 			success = false
 		}
+		if extensionWarning {
+			warning = true
+		}
 	}
 
 	return success, warning
```

Comment:
this is a slight behavior change for existing on-cluster checks. previously we were overwriting the value of `warning` on each extension check, so we were only returning the `warning` value for the last extension check to run. this has been fixed to return `warning`==`true` if any extension check warns.